src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/ConditionMatchSegment.cs

```diff
@@ -14,7 +14,16 @@ public ConditionMatchSegment(int index)
 
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return condMatch?.BackReference[_index].Value;
+            if ( condMatch.BackReferences != null)
+            {
+                //Url Rewrite
+                return condMatch.BackReferences.GetBackReferenceAtIndex(_index);
+            }
+            else
+            {
+                // For mod_rewrite
+                return condMatch?.BackReference[_index].Value;
```

Comment:
Null checking is inconsistent, and in this case it would have already nullref'ed above if condMatch was null